util/device_sw_utils/extract_sw_logs.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""Script to convert logs placed in a given section into a parse-friendly database.
+
+The tool uses the readelf utility to extract const string messages placed in a
+given section of the ELF file generated for the device sw application. It appends
+the offset address of each string constant with the base address of the section
+to get the compete address of the log. The SW log monitor looks up this database
+to extract the logs at the right address and prints it with the appropriate severity.
+"""
+
+import argparse
+import logging as log
+import os
+import re
+import subprocess
+import sys
+
+
+def extract_sw_logs(readelf, section, input, output):
+    # Get the base address of the section
+    addr = 0
+    cmd = "readelf -S " + input + " | grep -F " + section
```

Comment:
- avoid the grep here, you can do that in Python when iterating through the file.
- The path to "readelf" is passed as argument to this script, but it's not used here.

Reply:
Also, this command should be a list rather than a string (shell=True is EVIL).